src/aten/sycl/BatchNormKernels.cpp

```diff
@@ -0,0 +1,3965 @@
+  if (!canUse32BitIndexMath(self)) {
+    return Impl::General;
+  }
+
+  if (self.is_contiguous()) {
+    return self.strides()[1] == 1 ? Impl::ChannelsLast : Impl::Contiguous;
+  }
+
+  if (self.is_contiguous(at::MemoryFormat::ChannelsLast)) {
+    return Impl::ChannelsLast;
+  }
+
+  return Impl::General;
+}
+
+inline Impl batch_norm_choose_impl(const Tensor& in1, const Tensor& in2) {
+  auto imp1 = batch_norm_choose_impl(in1);
+  if (imp1 == Impl::General) {
+    return imp1;
+  }
+  auto imp2 = batch_norm_choose_impl(in2);
+  return imp1 == imp2 ? imp1 : Impl::General;
+}
+
+template <
+    typename scalar_t,
+    int64_t dim,
+    template <typename U> class PtrTraits = DefaultPtrTraits,
+    typename index_t = int64_t>
+static GenericPackedTensorAccessor<scalar_t, dim, PtrTraits, index_t>
+get_packed_accessor(const Tensor& t, c10::string_view var_name) {
+  constexpr auto expect_type = c10::CppTypeToScalarType<
+      typename std::remove_const<scalar_t>::type>::value;
+  const auto actual_type = t.scalar_type();
+  TORCH_CHECK(
+      actual_type == expect_type,
+      "Expected ",
+      var_name,
+      " to have type ",
+      expect_type,
+      " but got ",
+      actual_type);
+  return t.generic_packed_accessor<scalar_t, dim, PtrTraits, index_t>();
+}
+
+template <
+    typename scalar_t,
+    int64_t dim,
+    template <typename U> class PtrTraits = DefaultPtrTraits,
+    typename index_t = int64_t>
+static GenericPackedTensorAccessor<scalar_t, dim, PtrTraits, index_t>
+packed_accessor_or_dummy(const Tensor& t, c10::string_view var_name) {
+  if (!t.defined()) {
+    const std::array<index_t, dim> zeros{{0}};
+    return GenericPackedTensorAccessor<scalar_t, dim, PtrTraits, index_t>(
+        nullptr, zeros.data(), zeros.data());
+  }
+  return get_packed_accessor<scalar_t, dim, PtrTraits, index_t>(t, var_name);
+}
+
+struct InvStd {
+  template <typename T>
+  inline T operator()(T var, double epsilon) const {
+    T invstd = 0.0f;
+    if (var != static_cast<T>(0.0f) || epsilon != static_cast<T>(0.0f)) {
+      invstd = static_cast<T>(1.0f) / std::sqrt(var + static_cast<T>(epsilon));
+    }
+    return invstd;
+  }
+};
+
+struct Var {
+  template <typename T>
+  inline T operator()(T var, double epsilon) const {
+    return var;
+  }
+};
+
+int get_max_group_size(int simd = SIMD32) {
+  int max_size = syclMaxWorkGroupSize();
+  int shfl2_restricted_size = simd * simd;
+  return max_size > shfl2_restricted_size ? shfl2_restricted_size : max_size;
+}
+
+int get_num_threads(int nelem, int restricted_simd = SIMD32) {
+  int max_size = get_max_group_size(restricted_simd);
+  int thread_sizes[5] = {32, 64, 128, 256, max_size};
+  for (int i = 0; i < 5; ++i) {
+    if (nelem <= thread_sizes[i]) {
+      return thread_sizes[i];
+    }
+  }
+  return max_size;
+}
+
+int get_prefer_simd(int numPlane, int nHw) {
+  // decide SIMD: SIMD32 or SIMD16
+
+  auto dev_id = at::xpu::getDeviceIndexOfCurrentQueue();
```

Comment:
Remove the dev_id to align usages in other places. Using default argument in the following SYCL APIs, like getDeviceProperties, syclMaxWorkItemsPerEU and syclMaxWorkItemsPerTile. Default argument means current XPU device.